pkg/health/health_test.go

```diff
@@ -0,0 +1,129 @@
+package health
+
+import (
+	"testing"
+
+	. "github.com/anthonybishopric/gotcha"
+	"github.com/hashicorp/consul/api"
+	"github.com/square/p2/pkg/kp"
+)
+
+func TestFindWorst(t *testing.T) {
+	a := Result{
+		ID:     "testcrit",
+		Status: Critical,
+	}
+	b := Result{
+		ID:     "testwarn",
+		Status: Warning,
+	}
+	c := Result{
+		ID:     "testpass",
+		Status: Passing,
+	}
+
+	id, _ := FindWorst([]Result{a, b})
+	Assert(t).AreEqual(id, a.ID, "FindWorst between critical and warning should have returned testcrit")
+
+	id, _ = FindWorst([]Result{b, c})
+	Assert(t).AreEqual(id, b.ID, "FindWorst between warning and passing should have returned testwarn")
+
+	id, _ = FindWorst([]Result{c, c})
+	Assert(t).AreEqual(id, c.ID, "FindWorst between two passing results should have returned testpass")
+}
+
+func TestPickServiceResult(t *testing.T) {
+	t1 := mockServiceEntry("1", "passing")
+	t2 := mockServiceEntry("2", "passing")
+	t3 := mockServiceEntry("3", "passing")
+
+	// Catalog is healthy but KV is not present
+	testMap := getResult([]*api.ServiceEntry{t1, t2, t3})
+	catalog := []*api.ServiceEntry{t1, t2, t3}
+	res, err := selectResult(catalog, nil)
+	Assert(t).AreEqual(err, nil, "selectResult returned err")
```

Comment:
The whole point of these tests is (a) to make sure that your code behaves the way you think it should, (b) to make sure it stays that way as you rewrite it, and (c) when _a_ or _b_ break, to make it easy for you to understand what went wrong. Your change makes _c_ easier, so I'm happy.

(I don't remember this earlier conversation at all, btw.)